demands/__init__.py

```
@@ -130,10 +130,14 @@ def post_send(self, response, **kwargs):
         self._demand_success(response, kwargs)
         return response
 
-    def _demand_success(self, response, request_params):
+    def successful(self, response, request_params):
+        """Override this method to define what makes a successful response"""
         expected_codes = request_params.get('expected_response_codes', [])
+        return response.is_ok or response.status_code in expected_codes
```

I already read the intention but I think they should only provide a convenient default implementation of `is_successful`. If I define my own implementation they should still take priority. For example maybe 404 is acceptable but there are cases when the response should raise an error even if the code is 404 (much like right now I want to be able to raise an error even though the code is 200). Then I can override `is_successful` to `return some_special_condition() and super(...).is_successful()` and set `expected_response_codes=[404]`. With your suggestion I have to make sure I don't set `expected_response_codes` (a mistake that someone could easily make) and check the code for myself in my implementation of `is_successful`, which is a waste of reusable code.